xqueue_watcher/client.py

```diff
@@ -63,7 +63,14 @@ def _request(self, method, uri, timeout=0.5, **kwargs):
         r = None
         while not r:
             try:
-                r = self.session.request(method, url, auth=self.http_basic_auth, timeout=timeout, **kwargs)
+                r = self.session.request(
+                    method,
+                    url,
+                    auth=self.http_basic_auth,
+                    timeout=timeout,
+                    allow_redirects=False,
+                    **kwargs
+                )
```

Comment:
line 77/84 below is strange.  The assigned value of r isn't used anywhere, and no return value is sent on that code path (meaning None is sent, instead of the expected 2-tuple).  Any idea why that is, or if we should worry about it causing problems?